src/directives/toggle/toggle.directive.ts

```diff
@@ -65,21 +71,26 @@ export class IgxToggleDirective implements IToggleView, OnInit, OnDestroy {
         if (!this.collapsed) { return; }
 
         const player = this.animationActivation();
-        player.onStart(() => this.collapsed = !this.collapsed);
         player.onDone(() =>  {
             player.destroy();
             if (fireEvents) {
-                this.onOpen.emit();
+                this.onOpened.emit();
             }
         });
 
+        this.collapsed = !this.collapsed;
```

Comment:
I think collapsed should be changed when the animation completes, or at least _after_ onOpening as the -ing event is supposed to fire before any state is changed.

(this seems to be the case with `closing` and we should be consistent)

Reply:
Done.
I also changed a little how collapsed was set in open and close methods. In open I am setting it to false, not to !collapsed, and in close I am setting it to true. This allows me to have access to html elements inside the toggle box in onOpening event.